tests/integration/cases/shell_commands/test_shell_commands.py

```diff
@@ -0,0 +1,101 @@
+import os
+
+from twisted.internet import defer
+
+import manof
+import tests.integration
+
+
+class BasicCommandsTestCase(tests.integration.IntegrationTestCase):
+
+    @defer.inlineCallbacks
+    def test_run_and_rm(self):
+        self._logger.info('Testing run command happy flow')
+
+        image_name = 'run_test_image'
+
+        # sanity - removing containers if exists
+        yield self._remove_docker_container(image_name)
+
+        # run the image using manof
+        yield self._manof_command('run', [image_name])
+
+        # check the image exists
+        docker_log_output, _, _ = yield manof.utils.execute('docker logs {}'.format(image_name),
+                                                            cwd=None,
+                                                            quiet=False,
+                                                            logger=self._logger)
+
+        self.assertEqual(docker_log_output, image_name)
+
+        # remove the container using manof
+        yield self._manof_command('rm', [image_name])
+
+        # check the container doesn't exist exists
+        yield self.assertFailure(
+            manof.utils.execute('docker logs {}'.format(image_name),
+                                cwd=None,
+                                quiet=False,
+                                logger=self._logger),
+            manof.utils.CommandFailedError
+        )
+        self._logger.debug('Last command was supposed to fail')
+
+    @defer.inlineCallbacks
+    def test_provision_images(self):
+        self._logger.info('Testing provision images happy flow')
+
+        image_name = 'provision_test_image'
```

Comment:
I don't see what the redundancy of the images provides here (provision test image and lift test image are redundant imo) - I would stick to one since you're just checking different operations here. no need for custom flags or behavior (yet)
So keep it simple wins